lib/internal/url.js

```diff
@@ -1041,6 +1088,7 @@ class URL {
   }
 
   toJSON() {
+    this.#checkSearchParams();
```

Comment:
this function can just be `this.href`.

Reply:
Am I fine to also swap `toString` to using `this.href`, even though its technically defined after?

Reply:
toJSON and toString should call this.href, since technically they are `href`

Reply:
Switching these over to `return this.href` results in some WPT failures 🤔 

```
[UNEXPECTED_FAILURE][FAIL] URL interface: stringifier
assert_throws_js: calling stringifier with this = {} didn't throw TypeError function "function() {
            interfacePrototypeObject.toString.apply({}, []);
        }" did not throw
    at IdlInterface.<anonymous> (/Users/mattcowley/GitHub/nodejs/node/test/fixtures/wpt/resources/idlharness.js:2664:9)
    at Test.step (/Users/mattcowley/GitHub/nodejs/node/test/fixtures/wpt/resources/testharness.js:2599:25)
    at test (/Users/mattcowley/GitHub/nodejs/node/test/fixtures/wpt/resources/testharness.js:628:30)
    at self.subsetTestByKey (/Users/mattcowley/GitHub/nodejs/node/test/fixtures/wpt/resources/idlharness.js:44:14)
Command: /Users/mattcowley/GitHub/nodejs/node/out/Release/node  /Users/mattcowley/GitHub/nodejs/node/test/wpt/test-url.js 'idlharness.any.js'

[UNEXPECTED_FAILURE][FAIL] URL interface: operation toJSON()
assert_throws_js: calling operation with this = {} didn't throw TypeError function "function() {
            fn.apply(obj, args);
        }" did not throw
    at throwOrReject (/Users/mattcowley/GitHub/nodejs/node/test/fixtures/wpt/resources/idlharness.js:107:9)
    at IdlInterface.do_member_operation_asserts (/Users/mattcowley/GitHub/nodejs/node/test/fixtures/wpt/resources/idlharness.js:2405:9)
    at IdlInterface.<anonymous> (/Users/mattcowley/GitHub/nodejs/node/test/fixtures/wpt/resources/idlharness.js:2309:14)
    at Test.step (/Users/mattcowley/GitHub/nodejs/node/test/fixtures/wpt/resources/testharness.js:2599:25)
Command: /Users/mattcowley/GitHub/nodejs/node/out/Release/node  /Users/mattcowley/GitHub/nodejs/node/test/wpt/test-url.js 'idlharness.any.js'
```

Reply:
Because you need private property access to check if the current `this` is same as URL. I forgot about this!

Reply:
Thinking this through more, is this throwing a different exception now because it is trying to access a private method first (`#ensureSearchParamsUpdated`), instead of a private property `#context`? If so, is it fine to just update the test to expect this new exception?